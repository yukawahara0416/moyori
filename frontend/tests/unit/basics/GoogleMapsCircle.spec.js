import { shallowMount } from '@vue/test-utils'
import Component from '@/basics/GoogleMapsCircle.vue'

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    mapCenter: { lat: 35.68, lng: 139.76 }
  }
  wrapper = shallowMount(Component, {
    propsData,
    stubs: ['GmapCircle']
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('props', () => {
  it('mapCenter', () => {
    expect(wrapper.props().mapCenter).toStrictEqual(propsData.mapCenter)
    expect(wrapper.props().mapCenter instanceof Object).toBe(true)
    expect(wrapper.vm.$options.props.mapCenter.required).toBe(false)
  })
})
