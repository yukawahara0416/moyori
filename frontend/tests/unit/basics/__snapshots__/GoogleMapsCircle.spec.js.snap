// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<gmapcircle-stub
  center="[object Object]"
  options="[object Object]"
  radius="500"
/>
`;
